docs: add warning about spec

agoose77 opened the pull request:
This PR starts to add a few admonitions that indicate where syntax is not in the MyST spec. I think the idea here would be that over time we make MEPs to introduce this syntax, and we gradually remove the warnings :)

Thoughts?

rowanc1 commented on docs/cross-references.md:
``````diff
@@ -171,6 +171,12 @@ See [](#my-math-label) for an equation!
 
 `````{tip}
 # Labelling dollar math
+
+
+````{warning}
+The following syntax for labelling dollar-math equations is in `beta`, as it is not yet described by the [MyST Specification](https://myst-spec.readthedocs.io/en/latest/index.html). As such, it may change in the future.
+````
``````
I think that we might actually just want remove dollar math labels from the documentation, there are other ways to label that are more consistent (i.e. a directive or a target assignment `()=`)

agoose77 commented:
@rowanc1 I've updated the spec link to the internal myst-tools, and removed the mention of dollar-math labels entirely. Is that better?

BTW, feel free to edit as you see fit. I'll pop back to this in the near future.

rowanc1 commented:
Nice -- I think this is honestly pretty close.

There need to be a few other things that we should clean up in just removing the `(label)` from the docs. For example:

![image](https://user-images.githubusercontent.com/913249/223872293-4db02e7f-1540-4f55-87d8-f5e53b11dd04.png)

![image](https://user-images.githubusercontent.com/913249/223872621-dd7bd38b-a2b3-4019-8ba9-3b0fba2e1ceb.png)

I think updating the third example to use latex math labels still showcases the points well:

```
$$
\label{maxwell}
\begin{aligned}
\nabla \times \vec{e}+\frac{\partial \vec{b}}{\partial t}&=0 \\
\nabla \times \vec{h}-\vec{j}&=\vec{s}\_{e}
\end{aligned}
$$

$$ Ax=b \label{one-liner} $$

See [](#maxwell) for enlightenment and [](#one-liner) to do things on one line!
```

agoose77 commented:
![image](https://user-images.githubusercontent.com/1248413/223874717-aff764b8-19d2-4a3d-955f-60a381aebdff.png)

I'm going to log off and become a hermit crab.

rowanc1 commented:
I had to stare at that for a while to convince myself!!!

Thank you for the doc fixes, appreciate the help, looking forward to more in the future! 🚀

rowanc1 commented:
Live:
https://myst-tools.org/docs/mystjs/math#dollar-math-equations

Thank you!
